sql/plan.go

```diff
@@ -80,6 +80,8 @@ func (p *planner) makePlan(stmt parser.Statement) (planNode, error) {
 		return p.Select(n)
 	case *parser.Set:
 		return p.Set(n)
+	case *parser.SetTransaction:
+		return p.SetTransaction(n)
```

Comment:
I see. I think it is still nice to have the "transaction is not running" invariant be a panic in the client/txn layer. Let's have the invariant checked at the sql layer. This might mean having a IsRunning() call exported by client/txn.